rust/lance/src/io/exec/take.rs

```diff
@@ -108,6 +98,27 @@ impl Take {
             output_schema,
         }
     }
+
+    // manually implemented async for Send bound
+    #[allow(clippy::manual_async_fn)]
+    fn take_batch(
+        batch: RecordBatch,
```

Comment:
Should we just pass row_id array, which makes this function more self-contained.

Reply:
`row_ids` would have to be a reference to a column in `batch`, and it's much easier if all the parameters have `'static` lifetime.

I added a docstring which I think makes sense as a unit of work.